membership-attribute-service/app/controllers/PaymentUpdateController.scala

```diff
@@ -112,7 +112,10 @@ class PaymentUpdateController(commonActions: CommonActions, override val control
       createPaymentMethod = CreatePaymentMethod(
         accountId = subscription.accountId,
         paymentMethod = bankTransferPaymentMethod,
-        paymentGateway = account.paymentGateway.get, // this will need to change to use this endpoint for 'payment method' SWITCH
+        paymentGateway = account.paymentGateway
+          .getOrElse(
+            throw new RuntimeException(s"Unrecognised payment gateway for account ${subscription.accountId} for user $maybeUserId")
```

Comment:
I appreciate this was already throwing an exception here, but is there any value in making it produce a `Left` instead?

Reply:
I feel like the value would be that we could potentially handle it later, e.g. add the actual JSON/payment gateway raw value, to the error message further up the chain.
Also that the function signature of the function fully represents the possible results of running it, so easier to reason about/refactor in future without stray exceptions.